File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"
#include "rna_transcription.h"
#include "stdlib.h"
#include "string.h"


char *to_rna(const char *dna) {

    char *rna = (char *) malloc(strlen(dna) * sizeof(char));
    for (size_t x = 0; x < strlen(dna); x++) {
        switch (dna[x]) {
            case 'G':
                rna[x] = 'C';
                break;
            case 'C':
                rna[x] = 'G';
                break;
            case 'T':
                rna[x] = 'A';
                break;
            case 'A':
                rna[x] = 'U';
                break;
            default:
                rna[x] = dna[x];
                break;
        }
    }
    return rna;
}

File: solutions/java/rna-transcription/1/src/main/java/RnaTranscription.java
class RnaTranscription {

    String transcribe(String dnaStrand) {
        String transcripted = "";
        char[] dna = dnaStrand.toCharArray();

        for (int x = 0; x < dna.length; x++) {
            if (dna[x] == 'G') {
                transcripted += 'C';
            }
            if (dna[x] == 'C') {
                transcripted += 'G';
            }
            if (dna[x] == 'T') {
                transcripted += 'A';
            }
            if (dna[x] == 'A') {
                transcripted += 'U';
            }
        }

        return transcripted;
    }

}
